platform/pulp/platform/models/repository.py

```diff
@@ -1,10 +1,342 @@
+    type = models.TextField()
+    last_updated = models.DateTimeField(auto_now=True, blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+
+class Importer(Plugin):
+    """
+    A abstract content importer.
+
+    Fields:
+
+    :cvar feed_url: The URL of an external content source.
+    :type feed_url: models.TextField
+
+    :cvar validate: Validate the imported context.
+    :type validate: models.BooleanField
+
+    :cvar ssl_ca_certificate: A PEM encoded CA certificate used for authentication.
+    :type ssl_ca_certificate: models.TextField
+
+    :cvar ssl_client_certificate: A PEM encoded client certificate used for authentication.
+    :type ssl_client_certificate: models.TextField
+
+    :cvar ssl_client_key: A PEM encoded private key used for authentication.
+    :type ssl_client_key: models.TextField
+
+    :cvar ssl_validation: Indicates whether SSL peer validation must be performed.
+    :type ssl_validation: models.BooleanField
+
+    :cvar proxy_host: The hostname of an SSL proxy to be used.
+    :type proxy_host: models.TextField
+
+    :cvar proxy_port: The TCP port number of an SSL proxy to be used.
+    :type proxy_port: models.IntegerField
+
+    :cvar proxy_user: The user used to authenticate with an SSL proxy to be used.
+    :type proxy_user: models.TextField
+
+    :cvar proxy_password: The password used to authenticate with an SSL proxy to be used.
+    :type proxy_password: models.TextField
+
+    :cvar basic_auth_user: The user used in HTTP basic authentication.
+    :type basic_auth_user: models.TextField
+
+    :cvar basic_auth_password: The password used in HTTP basic authentication.
+    :type basic_auth_password: models.TextField
+
+    :cvar max_download_bandwidth: The max amount of bandwidth used per download.
+    :type max_download_bandwidth: models.IntegerField
+
+    :cvar max_concurrent_downloads: The number of concurrent downloads permitted.
+    :type max_concurrent_downloads: models.IntegerField
+
+    :cvar download_policy: The policy for downloading content.
+    :type download_policy: models.TextField
+
+    :cvar last_sync: When the last successful synchronization occurred.
+    :type last_sync: models.DateTimeField
+
+    Relations:
+
+    :cvar scratchpad: Arbitrary information stashed by the importer.
+    :type scratchpad: fields.GenericRelation
+    """
+
+    # Download Policies
+    IMMEDIATE = 'immediate'
+    ON_DEMAND = 'on_demand'
+    BACKGROUND = 'background'
+    DOWNLOAD_POLICIES = (
+        (IMMEDIATE, 'Download Immediately'),
+        (ON_DEMAND, 'Download On Demand'),
+        (BACKGROUND, 'Download In Background'))
+
+    feed_url = models.TextField()
+    validate = models.BooleanField(default=False)
+
+    ssl_ca_certificate = models.TextField(blank=True, default='')
+    ssl_client_certificate = models.TextField(blank=True, default='')
+    ssl_client_key = models.TextField(blank=True, default='')
+    ssl_validation = models.BooleanField(default=False)
+
```

Comment:
I think it would be good to include some more network configuration with a repository. The two things that come to mind are timeout and retry values. They could be based off the urllib3 objects (maybe not as granular, though). Maybe that's something we want to do later, but it's worth keeping in mind.

http://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#module-urllib3.util.timeout
http://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#module-urllib3.util.retry